plugins/colorschemes/base16/default.nix

```diff
@@ -37,94 +39,134 @@ helpers.vim-plugin.mkVimPlugin config {
+          customColorschemeType
+          helpers.nixvimTypes.rawLua
+        ];
+        default = literalMD ''`vim.env.BASE16_THEME` or `"schemer-dark"`'';
         description = ''
           The base16 colorscheme to use.
-          It can either be the name of a builtin colorscheme or an attrs specifying each color explicitly.
-
-          Example for the latter:
-          ```nix
-            {
-              base00 = "#16161D";
-              base01 = "#2c313c";
-              base02 = "#3e4451";
-              base03 = "#6c7891";
-              base04 = "#565c64";
-              base05 = "#abb2bf";
-              base06 = "#9a9bb3";
-              base07 = "#c5c8e6";
-              base08 = "#e06c75";
-              base09 = "#d19a66";
-              base0A = "#e5c07b";
-              base0B = "#98c379";
-              base0C = "#56b6c2";
-              base0D = "#0184bc";
-              base0E = "#c678dd";
-              base0F = "#a06949";
-            }
-          ```
+
+          You may use the name of a builtin colorscheme or an attrs that specifies the colors explicitly.
+
+          Examples of builtin themes include:
+          ${concatStrings (
+            map (e: ''
+              - "${e}"
+            '') builtinColorschemeExamples
+          )}
+
+          `:h nvim-base16-builtin-colorschemes` includes a full list of builtin themes,
+          however the [plugin's source code] may be more up to date.
+
+          [plugin's source code]: https://github.com/RRethy/base16-nvim/blob/master/lua/colors/init.lua
         '';
-        example = "edge-light";
+        example = customColorschemeExample;
       };
 
     setUpBar = mkOption {
       type = types.bool;
       default = true;
+      example = false;
       description = "Whether to set your status bar theme to 'base16'.";
     };
   };
 
-  extraConfig =
-    cfg:
-    mkMerge [
-      {
-        plugins.airline.settings.theme = mkIf cfg.setUpBar "base16";
-        plugins.lualine.theme = mkIf cfg.setUpBar "base16";
-        plugins.lightline.colorscheme = null;
-
-        opts.termguicolors = mkDefault true;
-      }
-      (mkIf (isString cfg.colorscheme) { colorscheme = "base16-${cfg.colorscheme}"; })
-      (mkIf (isAttrs cfg.colorscheme) {
-        extraConfigLuaPre = ''
-          require('base16-colorscheme').setup(${helpers.toLuaObject cfg.colorscheme})
-        '';
-      })
-    ];
+  # We will manually set the colorscheme, using `setup`
+  colorscheme = null;
+  callSetup = false;
+
+  extraConfig = cfg: {
+    plugins.airline.settings.theme = mkIf cfg.setUpBar (mkDefault name);
+    plugins.lualine.theme = mkIf cfg.setUpBar (mkDefault name);
+    plugins.lightline.colorscheme = mkDefault null;
+
+    opts.termguicolors = mkDefault true;
+    # FIXME: is it still ok to use `colorscheme` after using `setup`?
+    # It would be redundant, but maybe users want to read `config.colorscheme` in their nix config?
+    colorscheme = mkForce null;
```

Comment:
I don't think this needs a FIXME comment. TBH I'm not sure we need to set `colorscheme to `null` anyway, since that's the default.

If a user really wants to enable `colorschemes.base16` and still wants to call `:colorscheme` anyway, they've probably got a reason?